apps/managedidentity/managedidentity_test.go

```diff
@@ -454,6 +454,122 @@ func TestIMDSAcquireTokenReturnsTokenSuccess(t *testing.T) {
 	}
 }
 
+func TestCloudShellAcquireTokenReturnsTokenSuccess(t *testing.T) {
+	testCases := []struct {
+		resource string
+		miType   ID
+	}{
+		{resource: resource, miType: SystemAssigned()},
+		{resource: resourceDefaultSuffix, miType: SystemAssigned()},
+		{resource: resource, miType: UserAssignedClientID("clientId")},
+		{resource: resourceDefaultSuffix, miType: UserAssignedResourceID("resourceId")},
+		{resource: resourceDefaultSuffix, miType: UserAssignedObjectID("objectId")},
+	}
+	for _, testCase := range testCases {
+		t.Run(string(CloudShell)+"-"+testCase.miType.value(), func(t *testing.T) {
+			endpoint := os.Getenv(msiEndpointEnvVar)
```

Comment:
you need to set this, not get it. Otherwise, this test can only pass in Cloud Shell. At the moment this test actually panics when this variable isn't set